daemon/images/image_search_test.go

```diff
@@ -42,35 +43,40 @@ func TestSearchRegistryForImagesErrors(t *testing.T) {
 			shouldReturnError: true,
 		},
 		{
-			filtersArgs:   "invalid json",
-			expectedError: "invalid character 'i' looking for beginning of value",
-		},
-		{
-			filtersArgs:   `{"type":{"custom":true}}`,
+			filtersArgs:   filters.NewArgs(filters.Arg("type", "custom")),
```

Comment:
Do we already have some other good tests of `filters.FromJSON`, or would it be helpful to maybe include both here so we can _also_ use this list to test whether `filters.FromJSON` is generating what we expect from these test cases? :see_no_evil:

Reply:
There is a bunch of tests for `FromJSON`, but perhaps we should at some more (I was working on other changes in that package .. somewhere .. need to dig that up 😂)

Reply:
I rebased, and while add it, added checks in tests (both in `types/filters` and in `daemon/images`) to verify the returned error is also the correct type (not just contains the expected error message)